plenum/server/consensus/ordering_service.py

```diff
@@ -2075,3 +2070,49 @@ def l_processReqDuringBatch(self,
         if self.is_master:
             self.l_do_dynamic_validation(req, cons_time)
             self._write_manager.apply_request(req, cons_time)
+
+    def can_send_3pc_batch(self):
+        if not self._data.is_primary:
+            return False
+        if not self._data.is_participating:
+            return False
+        # ToDo: is pre_view_change_in_progress needed?
+        # if self.replica.node.pre_view_change_in_progress:
+        #     return False
+        if self.view_no < self.last_ordered_3pc[0]:
+            return False
+        if self.view_no == self.last_ordered_3pc[0]:
+            if self._lastPrePrepareSeqNo < self.last_ordered_3pc[1]:
+                return False
+            # This check is done for current view only to simplify logic and avoid
+            # edge cases between views, especially taking into account that we need
+            # to send a batch in new view as soon as possible
+            if self._config.Max3PCBatchesInFlight is not None:
+                batches_in_flight = self._lastPrePrepareSeqNo - self.last_ordered_3pc[1]
+                if batches_in_flight >= self._config.Max3PCBatchesInFlight:
+                    if self.l_can_log_skip_send_3pc():
+                        self._logger.info("{} not creating new batch because there already {} in flight out of {} allowed".
+                                          format(self.name, batches_in_flight, self._config.Max3PCBatchesInFlight))
+                    return False
+
+        self._skip_send_3pc_ts = None
+        return True
+
+    def l_can_log_skip_send_3pc(self):
+        current_time = time.perf_counter()
+        if self._skip_send_3pc_ts is None:
+            self._skip_send_3pc_ts = current_time
+            return True
+
+        if current_time - self._skip_send_3pc_ts > self._config.Max3PCBatchWait:
+            self._skip_send_3pc_ts = current_time
+            return True
+
+        return False
+
+    def can_order(self):
```

Comment:
Do we call this method properly?
Also I would prefer having this in validator class (just to break the monolith)